### ISSUE TITLE
Reports "no connection" instead of syncing

### ISSUE THREAD

**Issue by lunchboxer** [orgzly/orgzly-android#676]
- [x] I have searched for existing issues that may be the same as or related to mine.

When in China, my android device reports that I have no internet connection unless I am connected to a VPN. This is, however not accurate.  The only effect of this is that my statusbar includes a wifi icon with an x on it. Otherwise, I notice no ill effects... until now. There may be plenty of apps relying of the android system to report if the user has a usable internet connection, but I haven't noticed.

Expected behaviour: ignore what the android system says about the connection and sync, or try to sync and fail and then report the results appropriately

Actual behaviour: Orgzly refuses to start sync instead reporting "No connection"

Perhaps you don't consider this a bug. Great! re-categorize this issue as a feature request in that case. If it has to be that way then, I think it would be useful to have a simple toggle in the sync settings to ignore reported network conditions and boldly sync anyways.

**Comment by nevenz** [comment]
Currently `NET_CAPABILITY_VALIDATED` is checked for. Perhaps checking for `NET_CAPABILITY_INTERNET` would do it.

**Comment by lunchboxer** [comment]
Actually, the more I learn about this problem, the more I think that it's just a quirky behaviour at a very low level in the android system and expecting this app to fix it is a little silly. Being able to switch off the connection check for this one edge case would be kinda nice though

**Comment by 809825706** [comment]
I met this problem too, but I found it only happened in the system (mokee100 , havoc 3.5) which port from AOSP 10.  In MIUI 11 and oxygen 10, there is no need of VPN. Does the app need GMS to sync?

**Comment by 809825706** [comment]
I found sync succeed by force syncing, but it can only sync one file once.  So, is it possible to select multiple file to sync once? @nevenz

**Comment by nevenz** [comment]
> I met this problem too, but I found it only happened in the system (mokee100 , havoc 3.5) which port from AOSP 10. In MIUI 11 and oxygen 10, there is no need of VPN. Does the app need GMS to sync?

No, I think the issue is in the way NET_CAPABILITY_VALIDATED actually validates the connection. There are some clues in https://stackoverflow.com/q/62541728.

> I found sync succeed by force syncing, but it can only sync one file once. So, is it possible to select multiple file to sync by one click?

Not yet, multi-select for notebooks need to be implemented. Can you open a new issue for that? It will be useful for other actions as well.

**Comment by 809825706** [comment]
> > I met this problem too, but I found it only happened in the system (mokee100 , havoc 3.5) which port from AOSP 10. In MIUI 11 and oxygen 10, there is no need of VPN. Does the app need GMS to sync?
> 
> No, I think the issue is in the way NET_CAPABILITY_VALIDATED actually validates the connection. There are some clues in https://stackoverflow.com/q/62541728.
> 
> > I found sync succeed by force syncing, but it can only sync one file once. So, is it possible to select multiple file to sync by one click?
> 
> Not yet, multi-select for notebooks need to be implemented. Can you open a new issue for that? It will be useful for other actions as well.

I have opened a similar issue at #749.
